src/go/util/transport_socket_test.go

```diff
@@ -188,6 +189,42 @@ func TestCreateDownstreamTransportSocket(t *testing.T) {
 				}
 			} `,
 		},
+		{
+			desc:               "Downstream Transport Socket for mTLS",
+			sslPath:            "/etc/ssl/endpoints/",
+			sslRootCertPath:    "/etc/ssl/endpoints/root.crt",
+			sslMinimumProtocol: "TLSv1.1",
+			wantTransportSocket: `{
+				"name": "envoy.transport_sockets.tls",
+				"typedConfig": {
```

Comment:
Im not 100% familar with ESPv2 yet, but should there be a `require_client_certificate` flag in the config? https://www.envoyproxy.io/docs/envoy/latest/start/quick-start/securing

Reply:
https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/transport_sockets/tls/v3/common.proto#envoy-v3-api-msg-extensions-transport-sockets-tls-v3-certificatevalidationcontext And are we going to support all kind of client auth or just trustedCA?

Reply:
`require_client_certifacte` added. It is interesting that seems this field doesn't work and  client certificate  is enforced. As I didn't set it before, my test would raise error if the client doesn't provide certs, which fits https://github.com/envoyproxy/envoy/issues/17224.

Yes, this PR only provides basic client cert authn for this PR and for those [options](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/transport_sockets/tls/v3/common.proto#extensions-transport-sockets-tls-v3-certificatevalidationcontext), what do you need?

Reply:
im having a meeting with partner team on Wednesday and ill raise that point. but i guess this CL is a good start already, it should be easy to add new options in the future.

Reply:
SG